store/snap_remote_repo.go

```diff
@@ -244,6 +244,9 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string, auther Authentica
 	}
 
 	// set headers
+	if auther != nil {
+		auther.Authenticate(req)
+	}
 	s.applyUbuntuStoreHeaders(req, "")
```

Comment:
These should be factored out into a single method, something like:
func (s *SnapUbuntuStoreRepository) applyAuthorizedHeaders(req *http.Request, auther Authentication) {
  	s.applyUbuntuStoreHeaders(req, "")
	if auther != nil {
		auther.Authenticate(req)
	}
}